sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,200 @@ def transform(self, X):
+        count_cache_[X_i][y_i]
+
+    y_set_ : list of (index, y) tuples
+        An enumerated set of all unique values y can have
+
+    col_num_X_ : int
+        The number of columns of 'X' learned during 'fit'
+
+    col_num_Y_ : int
+        The number of clumns of 'y' learned during 'fit'
+        If 0, then the fit is not conditional on the y given
+
+    Examples
+    --------
+    Given a dataset with two features and four samples, we let the transformer
+    find the number of occurences of each data point in the dataset
+
+    >>> from sklearn.preprocessing.data import CountFeaturizer
+    >>> X = [[0], [0], [0], [0], [1], [1], [1], [1]]
+    >>> y = [0, 1, 1, 1, 0, 0, 0, 0]
+    >>> cf = CountFeaturizer()
+    >>> cf.fit_transform(X, y)      # doctest: +NORMALIZE_WHITESPACE
+    array([[ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.]])
+
+    See also
+    --------
+    https://blogs.technet.microsoft.com/machinelearning/2015/02/17/big-learning-made-easy-with-counts/      # noqa
+    """
+
+    def __init__(self, inclusion='all'):
+        self.inclusion = inclusion
+
+    @staticmethod
+    def _check_params(inclusion=None):
+        if (inclusion is not None and inclusion != 'all' and
+                not CountFeaturizer._valid_data_type(inclusion)):
+            raise ValueError("Illegal data type in inclusion")
+
+    @staticmethod
+    def _valid_data_type(type_check):
+        return type(type_check) == np.ndarray or type(type_check) == list \
+            or type(type_check) == set
+
+    def fit(self, X, y=None):
+        """Fits the CountFeaturizer to X, y
+
+        Stores the counts for each example X, conditional on y
+        Both X and y must be appropriately reshaped to a 2D list
+
+        Parameters
+        ----------
+        X : array
+            The data set to learn the counts from, conditional to 'y'
+
+        y : None, array
+            The 'y' that the counts are conditional to
+            - None (default) : The counts learned are not conditional on 'y'
+            - array : The counts learned are conditional on the given 'y'
+
+        Returns
+        -------
+        self
+        """
+
+        CountFeaturizer._check_params(inclusion=self.inclusion)
+
+        if y is not None:
+            X, y = check_X_y(X, y)
+            self.col_num_Y_ = len(y.take([0]))
+        else:
+            X = check_array(X)
+            self.col_num_Y_ = 0
+
+        len_data = len(X)
+
+        self.count_cache_ = defaultdict(lambda: defaultdict(int))
+        self.col_num_X_ = len(X[0])
+        self.y_set_ = set()
+
+        if type(self.inclusion) == str and self.inclusion == 'all':
+            inclusion_used = np.array(range(self.col_num_X_))
+        elif self.inclusion is None:
+            inclusion_used = np.array([])
```

Comment:
I thought it might be a strange edge case that might be handled. 
Do you think that it would be better to just throw an error when inclusion is None?